lib/facter/ipmitool.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+Facter.add(:ipmitool, type: :aggregate) do
+  # https://puppet.com/docs/puppet/latest/fact_overview.html
+  confine kernel: 'Linux'
+  confine is_virtual: false
+  # TODO: consider confining based on which
+  # this has the side affect that the ipmitool fact and ipmitool_mc_info facts would be Nil
+  # if ipmi is not present instead of the curent values of
+  # ipmitool: {"fru"=>{}, "mc_info"=>{"IPMI_Puppet_Service_Recommend"=>"stopped"}}
+  # ipmitool_mc_info: {"IPMI_Puppet_Service_Recommend"=>"stopped"}
+  # confine do
+  #   Facter::Util::Resolution.which('ipmitool')
+  # end
```

Comment:
Let me know if this is something worth pursuing or if you just want me to drop the comment.